### PR TITLE
Better clone everything

### PR DESCRIPTION
**Pull request by gurneyalex**
* use git fetch on already cloned repos
* try to use the github api to get an up to date list of repositories

### CODE REVIEW

File: tools/oca_projects.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 """
 
+from github_login import login
```
Comment (hbrunn) [inline tools/oca_projects.py]: shouldn't you anticipate an ImportError here?

File: tools/oca_projects.py [hunk 2]
```diff
@@ -107,9 +109,27 @@
...
-OCA_REPOSITORY_NAMES = []
-for repos in OCA_PROJECTS.itervalues():
-    OCA_REPOSITORY_NAMES += repos
+
+def get_repositories():
+    ignored = set([
+        'odoo-community.org',
+        'community-data-files',
+        'contribute-md-template',
+        'website',
```
Comment (hbrunn) [inline tools/oca_projects.py]: why?

File: tools/oca_projects.py (hunk 1 above)
Comment (lepistone) [inline tools/oca_projects.py]: Why? `ImportError` is good I think. I would handle specially only error messages which are misleading.

**Comment by lepistone** [comment]
:+1: thanks

File: tools/oca_projects.py (hunk 1 above)
Comment (hbrunn) [inline tools/oca_projects.py]: I understand the following that the script should still work if there's no github login, then it should fall back to the hardcoded list. The same should apply in my opinion if the module is not available

File: tools/oca_projects.py (hunk 2 above)
Comment (mdietrichc2c) [inline tools/oca_projects.py]: I think those OCA repositiories are not needed for every project.

Comment (yvaucher) [inline tools/oca_projects.py]: Those OCA repositories are not Odoo modules

**Comment by yvaucher** [comment]
:+1:

File: tools/oca_projects.py (hunk 2 above)
Comment (StefanRijnhart) [inline tools/oca_projects.py]: website and community-data files are.

**Pull request merged**